pkg/ephemeral/ephemeral_test.go

```diff
@@ -121,6 +148,13 @@ func (s *EphemeralSuite) TestContainerNameFilter(c *check.C) {
 	c.Check(filter.Filter(&corev1.Container{Name: "nomatch"}), check.Equals, false)
 }
 
+func (s *EphemeralSuite) TestPodNameFilter(c *check.C) {
+	filter := ephemeral.PodNameFilter("matches")
+
+	c.Check(filter.Filter(&corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "matches"}}), check.Equals, true)
+	c.Check(filter.Filter(&corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "nomatch"}}), check.Equals, false)
```

Comment:
optional: I'd do separate testcases...slightly more boilerplate, but for more structure in reporting...